synapse/http/proxyagent.py

```diff
@@ -64,7 +64,7 @@ def __init__(
         self,
         reactor,
         proxy_reactor=None,
-        contextFactory=BrowserLikePolicyForHTTPS(),
+        contextFactory: BrowserLikePolicyForHTTPS = None,
```

Comment:
This is the in wrong type, it should be `IPolicyForHTTPS`, I think.

Reply:
I'll also make it `Optional`